### PR TITLE
[cluster-admin] Highlighted list items for better UX

### PR DESCRIPTION
**Pull request by TilmanHaupt**
To ensure a similar user experience (UX), the following features are proposed:

- Minimum width column for the ResourceStatus
- Rows are now grey if the user hovers
- Rows are light grey if the detail view for the cluster is open
- If the user clicks the active list item the detail panel is closed.

### CODE REVIEW

**Review by edda — APPROVED** [review]
🙏

File: ui/cluster-admin/src/store.ts [hunk 1]
```diff
@@ -35,8 +35,8 @@ export interface State {
   }
   setClusterDetails: (cluster: Cluster | null) => void
   setClusterDetailPlugins: (plugins: Plugin[]) => void
-  showClusterDetails: boolean
-  setShowClusterDetails: (showClusterDetails: boolean) => void
+  showClusterDetails: string | undefined
```
Comment (uwe-mayer) [inline ui/cluster-admin/src/store.ts:38]: I would prefer to keep a boolean here for easier handling and reference.

File: ui/cluster-admin/src/components/ClusterListItem.tsx [hunk 2]
```diff
@@ -46,7 +54,12 @@ const ClusterListItem: React.FC<ClusterListItemProps> = (
   }
 
   return (
-    <DataGridRow style={{ cursor: "pointer" }} onClick={() => openDetails()}>
+    <DataGridRow
+      className={`cursor-pointer ${
+        showClusterDetails === props?.cluster?.metadata?.name ? "active" : ""
```
Comment (uwe-mayer) [inline ui/cluster-admin/src/components/ClusterListItem.tsx:59]: The `clusterDetails.cluster` should already provide the necessary data. No need to use `showClusterDetails` here.

File: ui/cluster-admin/src/components/ClusterDetail.tsx [hunk 3]
```diff
@@ -37,7 +37,7 @@ const ClusterDetail: React.FC<any> = () => {
   }, [clusterDetails.cluster?.metadata?.creationTimestamp])
 
   const onPanelClose = () => {
-    setShowClusterDetails(false)
+    setShowClusterDetails(undefined)
```
Comment (uwe-mayer) [inline ui/cluster-admin/src/components/ClusterDetail.tsx:40]: We need to also `null` the `cluster` prop in `clusterDetails` here to make sure the `active` state works correctly

**Review by uwe-mayer — CHANGES_REQUESTED** [review]

File: ui/cluster-admin/src/components/ClusterDetail.tsx (hunk 3 above)
Comment (uwe-mayer) [reply ui/cluster-admin/src/components/ClusterDetail.tsx:40]: So we need `setClusterDetails` from `useStore()` and set it here

**Comment by uwe-mayer** [comment]
Thanks for adding this! Just some remarks on how to extract `active` state without changing the `showClusterDetails` type

**Comment by TilmanHaupt** [comment]
> Thanks for adding this! Just some remarks on how to extract `active` state without changing the `showClusterDetails` type

Should be all fine now. :)

File: ui/cluster-admin/src/components/ClusterListItem.tsx [hunk 4]
```diff
@@ -46,7 +55,15 @@ const ClusterListItem: React.FC<ClusterListItemProps> = (
   }
 
   return (
-    <DataGridRow style={{ cursor: "pointer" }} onClick={() => openDetails()}>
+    <DataGridRow
+      className={`cursor-pointer ${
+        clusterDetails?.cluster?.metadata?.name ===
+          props?.cluster?.metadata?.name && showClusterDetails
```
Comment (uwe-mayer) [inline ui/cluster-admin/src/components/ClusterListItem.tsx:61]: Nice, no need to `null` the `clusterDetails.cluster`

**Review by uwe-mayer — APPROVED** [review]

**Pull request merged**